avm/res/network/virtual-network-gateway/main.bicep

```
@@ -494,6 +494,28 @@ output activeActive bool = virtualNetworkGateway.properties.activeActive
 @description('The location the resource was deployed into.')
 output location string = virtualNetworkGateway.location
 
+@description('The ASN (Autonomous System Number) of the virtual network gateway.')
+output asn int? = isBgp ? virtualNetworkGateway.properties.bgpSettings.asn : null //'Not applicable (No Bgp)'
```

This should be enough
```suggestion
output asn int? = virtualNetworkGateway.properties.bgpSettings.asn
```

---

In general instead of providing multiple top level properties I wonder if cleaner to provide a bgp object as output